src/sources/kubernetes_logs/parser/cri.rs

```diff
@@ -107,49 +119,56 @@ impl FunctionTransform for Cri {
                         })
                     });
 
-                    let mut drop_original = true;
                     for (name, value) in captures {
-                        // If we're already overriding the original field, don't remove it after.
-                        if name == self.field {
-                            drop_original = false;
+                        match name.as_str() {
+                            MESSAGE_KEY => {
+                                // Insert either directly into `.` or `log_schema().message_key()`,
+                                // overwriting the original "full" CRI log that included additional fields.
+                                drop(log.insert(message_field, value));
+                            }
+                            MULTILINE_KEY => {
+                                // If the MULTILINE_TAG is 'P' (partial), insert our generic `_partial` key.
+                                // This is safe to `unwrap()` as we've just ensured this value is a Value::Bytes
+                                // during the above capturing and mapping.
+                                if value.as_bytes().unwrap()[0] == b'P' {
+                                    self.log_namespace.insert_source_metadata(
+                                        Config::NAME,
+                                        log,
+                                        Some(LegacyKey::Overwrite(path!(event::PARTIAL))),
+                                        path!(event::PARTIAL),
+                                        true,
+                                    );
+                                }
+                            }
+                            TIMESTAMP_KEY => {
+                                // Insert the TIMESTAMP_TAG parsed out of the CRI log, this is the timestamp of
+                                // when the runtime processed this message.
+                                self.log_namespace.insert_source_metadata(
+                                    Config::NAME,
+                                    log,
+                                    Some(LegacyKey::Overwrite(path!(log_schema().timestamp_key()))),
+                                    path!(TIMESTAMP_KEY),
+                                    value,
+                                );
+                            }
+                            STREAM_KEY => {
+                                self.log_namespace.insert_source_metadata(
+                                    Config::NAME,
+                                    log,
+                                    Some(LegacyKey::Overwrite(path!(STREAM_KEY))),
+                                    path!(STREAM_KEY),
+                                    value,
+                                );
+                            }
+                            _ => {
+                                unreachable!("all CRI captures groups should be matched");
```

Comment:
Just wondering about the future here- is there a possibility that the set of names provided from the captures, changes in the future, and if so, do we have proper automated tests to ensure that this code would need to be updated?
Otherwise, feel free to drop it if it's not relevant.

Reply:
Yep, this should fail in the `parser` tests both at the picker and the cri level. I think longer term this should be moved out into the `codec` library though.